nixos/modules/config/console.nix

```diff
@@ -139,8 +147,13 @@ in
         environment.etc.kbd.source = "${consoleEnv pkgs.kbd}/share";
 
         boot.initrd.preLVMCommands = mkIf (!config.boot.initrd.systemd.enable) (mkBefore ''
-          kbd_mode ${if isUnicode then "-u" else "-a"} -C /dev/console
-          printf "\033%%${if isUnicode then "G" else "@"}" >> /dev/console
+          if ${isUnicode} ; then
```

Comment:
Is the executing environment here setup with the locales from `i18n.extraLocaleSettings`? If so that could be an issue:
```
 env -i LANG=eo LC_IDENTIFICATION=en_US LOCALE_ARCHIVE=(nix eval --raw self#glibcLocales)/lib/locale/locale-archive locale -k identification-codeset
identification-codeset="ISO-8859-1"
```